test/token/ERC20/ERC20.test.js

```diff
@@ -1,34 +1,44 @@
-const { BN, constants, expectEvent, expectRevert } = require('@openzeppelin/test-helpers');
+const { ethers } = require('hardhat');
 const { expect } = require('chai');
-const { ZERO_ADDRESS } = constants;
+const { loadFixture } = require('@nomicfoundation/hardhat-network-helpers');
+const { PANIC_CODES } = require('@nomicfoundation/hardhat-chai-matchers/panic');
 
 const {
-  shouldBehaveLikeERC20,
-  shouldBehaveLikeERC20Transfer,
-  shouldBehaveLikeERC20Approve,
+  bigint: { shouldBehaveLikeERC20, shouldBehaveLikeERC20Transfer, shouldBehaveLikeERC20Approve },
 } = require('./ERC20.behavior');
-const { expectRevertCustomError } = require('../../helpers/customError');
 
-const TOKENS = [
-  { Token: artifacts.require('$ERC20') },
-  { Token: artifacts.require('$ERC20ApprovalMock'), forcedApproval: true },
-];
+const TOKENS = [{ Token: '$ERC20' }, { Token: '$ERC20ApprovalMock', forcedApproval: true }];
 
-contract('ERC20', function (accounts) {
-  const [initialHolder, recipient] = accounts;
+const name = 'My Token';
+const symbol = 'MTKN';
+const initialSupply = 100n;
 
-  const name = 'My Token';
-  const symbol = 'MTKN';
-  const initialSupply = new BN(100);
+async function fixture() {
+  const [initialHolder, recipient, anotherAccount] = await ethers.getSigners();
+
+  const tokens = {};
+  for (const { Token } of TOKENS) {
+    const token = await ethers.deployContract(Token, [name, symbol]);
+    await token.$_mint(initialHolder, initialSupply);
+
+    tokens[Token] = token;
+  }
+
+  return { initialHolder, recipient, anotherAccount, tokens };
+}
+
+describe('ERC20', function () {
+  beforeEach(async function () {
+    Object.assign(this, await loadFixture(fixture));
+  });
 
   for (const { Token, forcedApproval } of TOKENS) {
```

Comment:
I would move the `fixture` function inside this, so that only one contract is deployed.

Something like
```javascript
for (const { Token, forcedApproval } of TOKENS) {
  async function fixture() {
    // ...
    const token = await ethers.deployContract(Token, [name, symbol]);
    await token.$_mint(initialHolder, initialSupply);
  
    return { /* ... */, token };
  }

  beforeEach(async function () {
    Object.assign(this, await loadFixture(fixture));
  });
  
  shouldBehaveLikeERC20(initialSupply, { forcedApproval });
  
  // ...
}

Reply:
What @ernestognw was to have only one fixture per file to increase the optimization

Reply:
Its between:
- one fixture, executed one time, that deploys X contracts
- X fixtures, executed one time each, that deploys 1 contracts each.

Number of deployment is the same... but the second option does more snapshots (each feature comes with its own snapshot).

I like that the second option sets `this.token` correctly without having to rely on an additional `beforeEach`. I also like the fact that it makes the fixture code simpler. I'm not sure what the performance difference are.

Reply:
I think is generally better to have a single fixture for 2 reasons:

1. Seems like [fixtures are identified by comparing the function with a list of fixtures](https://github.com/NomicFoundation/hardhat/blob/c3aad2c556a4825d53667af40a065a350e5317be/packages/hardhat-network-helpers/src/loadFixture.ts#L38). My understanding of this is that multiple fixtures slow down execution if there are lots of them.
2. Usually fixtures differ in 1 or 2 contracts (i.e. deploy the same with exception of one or two contracts), my conclusion is that it's better to have 1 single fixture with 5 contracts deployed than 2 fixtures with 4 contracts deployed.

Number 1 is only a problem if we have multiple fixtures, but we only have 2, and Number 2 is just a small preference. No strong opinion on this one, both are fine.

Reply:
See if you approve the new version